pkg/joinserver/redis/registry.go

```diff
@@ -160,6 +160,9 @@ func (r *DeviceRegistry) set(tx *redis.Tx, uid string, gets []string, f func(pb
 	if stored == nil && pb == nil {
 		return nil, nil
 	}
+	if pb != nil && len(sets) == 0 {
+		return stored, nil
```

Comment:
Why not `pb = stored` like in all other Redis registries? We should keep this identical

Reply:
Because this is a different kind of function, this one returns the result and error, whilst others return just error.
We still have to return something here, sure we could make it
```go
pb = stored
return pb, nil
```
But that's redundant, since we can just `return stored, nil`